models/workday_history/workday__employee_daily_history.sql

```diff
@@ -0,0 +1,80 @@
+-- depends_on: {{ ref('int_workday__employee_history') }}
+{{ config(enabled=var('employee_history_enabled', False)) }}
+
+{% if execute %} 
+    {% set first_last_date_query %}
+    with min_max_values as (
+
+        select 
+            min(_fivetran_start) as min_start,
+            max(_fivetran_start) as max_start 
+        from {{ ref('int_workday__employee_history') }}
+    )
+
+    select 
+        min_start,
+        case when max_start >= {{ dbt.current_timestamp() }}
+            then max_start
+            else {{ dbt.date_trunc('day', dbt.current_timestamp()) }} 
+        end as max_start
+    from min_max_values
+    
+    {% endset %}
+
+    {% set start_date = run_query(first_last_date_query).columns[0][0]|string %}
+    {% set last_date = run_query(first_last_date_query).columns[1][0]|string %}
+
+{# If only compiling, creates range going back 1 year #}
+{% else %} 
+    {% set start_date = dbt.dateadd("year", "-2", "current_date") %} -- Arbitrarily picked. Choose a more appropriate default if necessary.
+    {% set last_date = dbt.dateadd("year", "-1", "current_date") %}
```

Comment:
To match the comment above, let's just make this range truly be for one year. This is not necessary as it will not be used by a customer and only for compilation, but I want to ensure we don't get in trouble with causing unnescary confusion down the line.
```suggestion
    {% set start_date = dbt.dateadd("year", "-1", "current_date") %} -- One year in the past for first date
    {% set last_date = dbt.dateadd("day", "-1", "current_date") %} -- Yesterday as last date
```

Reply:
Committed.